app/scripts/controllers/landing.js

```diff
@@ -28,26 +28,28 @@ angular.module('confRegistrationWebApp')
     ];
     $scope.jumboImg = jumboImgs[_.random(0, jumboImgs.length - 1)];
 
-    $scope.eventSearch = _.throttle(function(val) {
+    $scope.eventSearch = _.debounce(function(val) {
       if(!val){ return; }
       if(val.length < 2){
         return;
       }
-      $http.get('conferences', {
-        params: {
-          conferenceName: val
-        }
-      }).then(function(response){
-        $scope.eventFilters = {
-          locationName: '',
-          date: ''
-        };
-        $scope.searchComplete = val;
-        $scope.searchResults = response.data;
+      if(angular.isDefined($scope.searchVal) && val === $scope.searchVal) {
```

Comment:
So at this point, I think essentially no time has passed and this will always be true. Where we run into problems is in the `then` handler below where a request has been in flight for a while and may finish out of order. It is there we want to check to make sure the response is answering the correct search query before showing that response to the user.

If you want to try and abort the previous request too you could look into using the `timeout` config https://docs.angularjs.org/api/ng/service/$http#usage.